datajoint/fetch.py

```diff
@@ -20,46 +25,57 @@ def prepare_attributes(relation, item):
         raise DataJointError("Index must be a slice, a tuple, a list, a string.")
     return item, attributes
 
-class FetchQuery:
+def copy_first(f):
+    @wraps(f)
+    def ret(*args, **kwargs):
+        args = list(args)
+        args[0] = args[0].__class__(args[0]) # call copy constructor
+        return f(*args, **kwargs)
 
-    def __init__(self, relation):
-        """
+    return ret
 
-        """
-        self.behavior = dict(
-            offset=0, limit=None, order_by=None, descending=False, as_dict=False, map=None
-        )
-        self._relation = relation
+class Fetch:
+    def __init__(self, relation):
+        if isinstance(relation, Fetch): # copy constructor
+            self.behavior = dict(relation.behavior)
+            self._relation = relation._relation
+        else:
+            self.behavior = dict(
+                offset=0, limit=None, order_by=None, as_dict=False
+            )
+            self._relation = relation
 
 
+    @copy_first
     def from_to(self, fro, to):
         self.behavior['offset'] = fro
         self.behavior['limit'] = to - fro
         return self
 
-    def order_by(self, order_by):
-        self.behavior['order_by'] = order_by
+    @copy_first
+    def order_by(self, *args):
+        if len(args) > 0:
+            self.behavior['order_by'] = self.behavior['order_by'] if self.behavior['order_by'] is not None else []
+            namepat = re.compile(r"\s*(?P<name>\w+).*")
+            for a in args: # remove duplicates
```

Comment:
If SQL sorts from last to first, this means that the first attribute is the major sort key and the others are subordinate keys. The question is, what do we expect from a call `rel.fetch.order_by('attr1').order_by('attr2')`? Is attr1 the major sort key or the subordinate? 

- If attr1 is subordinate, we can just insert new order keys in front of `behavior['order_by']` and don't even need to remove duplicates. Removing them would probably be a bit more efficient on MySQL side, since I don't expect that MySQL checks whether the keys are double. 

- If attr1 is supposed to be the major sort key, then we need to agree on whether a newly added attribute with the same name should cancel out the old one or not. I think I am actually slightly in favour of not removing it and implementing `rel.fetch.order_by(None)` to delete the current order_by chain.